### DISCUSSION TITLE
matrix error in MARSS_tmb

### DISCUSSION THREAD

**Discussion by emmats** [atsa-es/fish550-2023#35] [Q&A]
@eeholmes I got this error for the following code. Do I need to update something on my end?: 
dfa.fit1<-MARSS_tmb(tsdat, model=model.list, z.score=T, form='dfa', control=cntl.list)

Warning message:
In checkMatrixPackageVersion() : Package version inconsistency detected.
TMB was built with Matrix version 1.5.1
Current Matrix version is 1.4.0
Please re-install 'TMB' from source using install.packages('TMB', type = 'source') or ask CRAN for a binary version of 'TMB' matching CRAN's 'Matrix' package

**Comment by eeholmes** [comment]
Hi Emma,

Can you post the full code so I can see model.list and cntl.list?

Eli

**Comment by eeholmes** [comment]
Ah wait the error has to do with your Matrix package version being different than TMB.

Try this 
```
packageVersion("Matrix")
```
and then try updating Matrix so that the Matrix version is earlier than 1.5.0

**Reply by emmats** [reply]
I think the problem is that my Matrix is 1.4 and yours is 1.5. update.packages("Matrix") doesn't change my version. I tried to reinstall Matrix with install.packages, which required compilation, which failed at the very end so I'm still stuck with 1.4.

clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Matrix.so] Error 1
ERROR: compilation failed for package ‘Matrix’
* removing ‘/Library/Frameworks/R.framework/Versions/4.1/Resources/library/Matrix’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/4.1/Resources/library/Matrix’
Warning in install.packages :
  installation of package ‘Matrix’ had non-zero exit status

**Comment by emmats** [comment]
bact.ln<-log(bact.only2)
spd <- 6 # samples per day; every 4 hours
tsdat <- ts(bact.ln, freq=spd)
model.list<-list(m=1, R="diagonal and equal")
cntl.list<-list(maxit=100)
dfa.fit1<-MARSS_tmb(tsdat, model=model.list, z.score=T, form='dfa', control=cntl.list)

**Comment by eeholmes** [comment]
What sort of computer are you on?

On Fri, May 5, 2023 at 10:12 AM emmats ***@***.***> wrote:

> I think the problem is that my Matrix is 1.4 and yours is 1.5.
> update.packages("Matrix") doesn't change my version. I tried to reinstall
> Matrix with install.packages, which required compilation, which failed at
> the very end so I'm still stuck with 1.4.
>
> clang: error: linker command failed with exit code 1 (use -v to see
> invocation)
> make: *** [Matrix.so] Error 1
> ERROR: compilation failed for package ‘Matrix’
>
>    - removing
>    ‘/Library/Frameworks/R.framework/Versions/4.1/Resources/library/Matrix’
>    - restoring previous
>    ‘/Library/Frameworks/R.framework/Versions/4.1/Resources/library/Matrix’
>    Warning in install.packages :
>    installation of package ‘Matrix’ had non-zero exit status
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/atsa-es/fish550-2023/discussions/35#discussioncomment-5818378>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AATNSOV2UYQHDEUAK7NRAD3XEUYILANCNFSM6AAAAAAXXLRTIY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
------------------------------------
EE Holmes, NOAA Fisheries, Seattle, WA   206-860-3369 (or 206-419-6164)

**Reply by emmats** [reply]
macbook pro

**Comment by eeholmes** [comment]
Ah wait, are you using R 4.1? I think that's why you can't update to the
latest Matrix.

On Fri, May 5, 2023 at 10:32 AM Elizabeth Holmes - NOAA Federal <
***@***.***> wrote:

> What sort of computer are you on?
>
> On Fri, May 5, 2023 at 10:12 AM emmats ***@***.***> wrote:
>
>> I think the problem is that my Matrix is 1.4 and yours is 1.5.
>> update.packages("Matrix") doesn't change my version. I tried to reinstall
>> Matrix with install.packages, which required compilation, which failed at
>> the very end so I'm still stuck with 1.4.
>>
>> clang: error: linker command failed with exit code 1 (use -v to see
>> invocation)
>> make: *** [Matrix.so] Error 1
>> ERROR: compilation failed for package ‘Matrix’
>>
>>    - removing
>>    ‘/Library/Frameworks/R.framework/Versions/4.1/Resources/library/Matrix’
>>    - restoring previous
>>    ‘/Library/Frameworks/R.framework/Versions/4.1/Resources/library/Matrix’
>>    Warning in install.packages :
>>    installation of package ‘Matrix’ had non-zero exit status
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/atsa-es/fish550-2023/discussions/35#discussioncomment-5818378>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AATNSOV2UYQHDEUAK7NRAD3XEUYILANCNFSM6AAAAAAXXLRTIY>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>
>
> --
> ------------------------------------
> EE Holmes, NOAA Fisheries, Seattle, WA   206-860-3369 (or 206-419-6164)
>


-- 
------------------------------------
EE Holmes, NOAA Fisheries, Seattle, WA   206-860-3369 (or 206-419-6164)

**Reply by emmats** [reply]
Yes! Do I need to update my R Studio?

**Comment by eeholmes** [comment]
No, I don't think so. Just R but I usually do R and RStudio together just
to get that updating pain done in one step.

Once you install R, make sure to update all your packages. A lot of them
will need updating since like Matrix, many will have been pegged to an
older R version.

On Fri, May 5, 2023 at 10:35 AM emmats ***@***.***> wrote:

> Yes! Do I need to update my R Studio?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/atsa-es/fish550-2023/discussions/35#discussioncomment-5818543>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AATNSOWD57QCAMK6VHQZBQLXEU25VANCNFSM6AAAAAAXXLRTIY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
------------------------------------
EE Holmes, NOAA Fisheries, Seattle, WA   206-860-3369 (or 206-419-6164)

**Reply by emmats** [reply]
That worked. Thanks!
